src/components/authentication/index.tsx

```diff
@@ -71,39 +73,22 @@ export class Container extends React.Component<Properties, State> {
       !this.props.currentAddress &&
       this.props.user.data !== null
     ) {
-      this.logout();
+      this.props.clearSession();
     }
   }
 
-  logout(): void {
-    this.props.clearSession();
-  }
-
-  authorize(): void {
-    const web3Provider = this.props.providerService.get();
-
-    const method = 'personal_sign';
-    const from = Web3Utils.toHex(this.props.currentAddress.toLowerCase());
-    const params = [
-      config.web3AuthenticationMessage,
-      from,
-    ];
-
-    web3Provider.provider.sendAsync(
-      {
-        method,
-        params,
-        from,
-      },
-      (err, res) => {
-        if (err) {
-          this.props.updateConnector(Connectors.None);
-          return;
-        }
-
-        this.props.nonceOrAuthorize({ signedWeb3Token: res.result });
-      }
-    );
+  async authorize() {
+    const { personalSignToken, currentAddress, nonceOrAuthorize, updateConnector, clearSession, providerService } =
+      this.props;
+
+    await personalSignToken(providerService.get(), currentAddress)
+      .then((signedWeb3Token) => {
+        clearSession();
```

Comment:
@hkhanouchi, I added a call to `clearSession` to reset the user state and chat access token before we authorize a new user. You can view the `clearSession` logic [here](https://github.com/zer0-os/zOS/blob/main/src/store/authentication/saga.ts#L36).